### PR TITLE
Added logic to delete old profile image as user add new profile image 

### LINKED ISSUE #200: Improve Profile Image storing logic to manage redundancy of profile images in Storage
### 🚀 The feature

As I mentioned in the meeting, when the user changes the profile image it is added to storage without deleting the old profile image, and it increases the number of redundant images in Storage. The old profile image is of no use and we should delete it.
But the problem is that the profile image is stored using an auto-generated Unique ID, and as we don't know that ID, it is not possible to delete old images from storage. 

## Screenshot of Problem
image
We can see here that old profile images are not deleted from Storage

### Motivation, pitch

So, I propose that we create a profile image ID by email or username or any other method, but the IDs will be known to us so we can delete the old image. And reduce redundancy.

@chandansgowda 
I want to work on this. Please can you assign this task to me?

### PR DESCRIPTION
**Pull request by vrundraval24**
Fixes #200 

## Description

I added logic to delete the old profile image of the user from storage as the user change or add a new profile image. This way we can maintain the usage of Storage and reduce the redundancy of profile images.

I tried other ways like storing image using user uid or email but it didn't work well so at last,

I created a new attribute in User Docs in the Database to store the UserProfileImageID and used it to delete the Old user image. 
I used DateTime millieSecondsSinceEpoch as a Unique ID for the profile image.

image

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Everything at the front end and the database works fine. Shared some screenshots and screen-recording to review the changes.

## Screen recordings

https://github.com/AOSSIE-Org/Resonate/assets/141143426/0ba369ed-39f9-49d7-98cf-7a76336e4568

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #200 
- [x] Tag the PR with the appropriate labels

@chandansgowda Please review this.

### CODE REVIEW

File: lib/controllers/edit_profile_controller.dart [hunk 1]
```diff
@@ -171,10 +172,21 @@ class EditProfileController extends GetxController {
...
+        try {
+          await storage.deleteFile(
+              bucketId: userProfileImageBucketId,
+              fileId: authStateController.profileImageID!);
+        } catch (e) {
+          log(e.toString());
+        }
+
+        uniqueIdForProfileImage =
+            DateTime.now().millisecondsSinceEpoch.toString();
```
Comment (chandansgowda) [inline lib/controllers/edit_profile_controller.dart:184]: Don't you think in some cases that two people can have the same profileimage id ?

File: lib/controllers/auth_state_controller.dart [hunk 2]
```diff
@@ -68,7 +70,7 @@ class AuthStateController extends GetxController {
 
   Future<String> getAppwriteToken() async {
     Jwt authToken = await account.createJWT();
-    print(authToken);
+    // print(authToken);
```
Comment (chandansgowda) [inline lib/controllers/auth_state_controller.dart:73]: Let's change this to log or just delete the line.

**Comment by vrundraval24** [comment]
@chandansgowda The Requested changes are made. Please review them.

**Comment by chandansgowda** [comment]
Great job @vrundraval24 🙌

**Pull request merged**